editor/effects.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+import { parse, stringify } from 'querystring';
+
+/**
+ * WordPress dependencies
+ */
+import { getBlockSettings, switchToBlockType } from 'blocks';
+
+/**
+ * Internal dependencies
+ */
+import { focusBlock, replaceBlocks } from './actions';
+
+export default {
+	REQUEST_POST_UPDATE( action, store ) {
+		const { dispatch } = store;
+		const { postId, edits } = action;
+		const isNew = ! postId;
+		const toSend = postId ? { id: postId, ...edits } : edits;
+
+		new wp.api.models.Post( toSend ).save().done( ( newPost ) => {
+			dispatch( {
+				type: 'REQUEST_POST_UPDATE_SUCCESS',
+				post: newPost,
+			} );
+		} ).fail( ( err ) => {
+			dispatch( {
+				type: 'REQUEST_POST_UPDATE_FAILURE',
+				error: get( err, 'responseJSON', {
```

Comment:
I think I am following you, please correct me if I am missing the mark (I often do 😅 )  You are concerned over the coupling of the request success action to how it is being used in the reducer? In my apps I usually have a separate action like INSERT_{ENTITY}_INTO_{STORE} or when deleting REMOVE_{ENTITY}_FROM_{STORE}.  That can serve multipurpose cases rather than just network.  These are often triggered by the success/failure network actions.  They work well to solve what I think you are bringing up as a problem. I might not understand though, so hopefully that helps. Would love to hear how you handle this as well, or plan to.